### ISSUE TITLE
Ignore @NotNull annotations

### ISSUE THREAD

**Issue by nvervelle** [swagger-api/swagger-core#4213]
In [ModelResolver](https://github.com/swagger-api/swagger-core/blob/master/modules/swagger-core/src/main/java/io/swagger/v3/core/jackson/ModelResolver.java#L105), the list `NOT_NULL_ANNOTATIONS` is defined and contains the annotations that implies that the field is required.

Is there a way to disable this behavior, or remove some annotations from the list ? Or to ignore the annotation on some attributes ?

The assumption that a field annotated `@NotNull` should always be declared as `required = true` is incorrect and we should be able to disable this behavior. It can cause problem both for incoming data structure and for outgoing data structure.

### Example of problems for incoming data structure

```java
    @Schema(description = "Additional properties")
    @JsonProperty("additionalProperties")
    @JsonSetter(nulls = Nulls.AS_EMPTY)
    @NotNull
    Map<String, String> additionalProperties
```
The parameter `additionalProperties` should be marked as `required = false`, even with the `@NotNull` annotation : when the `additionalProperties` attribute is not provided, the `@JsonSetter` annotation will ensure that the parameter is properly initialized with an empty Map. So on the Java side, it's clearly a non-nullable attribute, but on the openapi spec side it should be `required = false`.

### Example of problems for outgoing data structure

```java
    @Schema(description = "Additional properties")
    @JsonProperty("additionalProperties")
    @JsonInclude(JsonInclude.Include.NON_EMPTY)
    @NotNull
    Map<String, String> additionalProperties
```
The parameter `additionalProperties` should be marked as `required = false`, even with the `@NotNull` annotation : when the `additionalProperties` parameter is an empty map, the `@JsonInclude` annotation will ensure that the attribute is not added to the payload. So on the Java side, it's a non-nullable attribute, but on the openapi spec side it should be `required = false`.

**Comment by nvervelle** [comment]
One idea to allow disabling the `NOT_NULL_ANNOTATIONS` on a given attribute : 
* Add a attribute `forceUseRequired` on `@Schema` annotation, with a default value as `false`
* Ignore the `NOT_NULL_ANNOTATIONS` when `forceUseRequired` is equal to `true`

I can submit a simple PR for this proposal.

**Comment by frantuma** [comment]
Thanks for looking into it, and yes a PR is welcome

A possible alternative to your proposal is still using a new `@Schema` field but also deprecating `required`, similar to what done with [AccessMode](https://github.com/swagger-api/swagger-core/blob/364932726b2b9f2e00db763ccd5ed4a1d0ddd270/modules/swagger-annotations/src/main/java/io/swagger/v3/oas/annotations/media/Schema.java#L230=)

This would have the advantage of having a single field to define the behavior, something like:

`RequiredMode requiredMode() default RequiredMode.AUTO;`

with

```
    enum RequiredMode {
        AUTO,
        REQUIRED,
		FORCED_REQUIRED
        NOT_REQUIRED,
    }
```

**Comment by nvervelle** [comment]
Hi @frantuma, thanks for the clarification.

I've submitted a second PR #4221 implementing your suggestion.

**Comment by nvervelle** [comment]
Hi @frantuma , any hope of validating the PR ? https://github.com/swagger-api/swagger-core/pull/4221

**Comment by frantuma** [comment]
Thanks a lot! merged #4286 which includes #4221
